src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,14 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer {
```

Comment:
The MachineProducer interface should be parameterized to ensure that only `Machine` or its subtypes can be used. Update the interface implementation to use a generic type parameter.